### ISSUE TITLE
Invalid object name 'dbo.TokenCacheEntries'.

### ISSUE THREAD

**Issue by codymullins** [Azure-Samples/AzureAD-WebApp-WebAPI-OAuth2-UserIdentity-DotNet#4]
I pulled the project, made the changes, and ran it.

After creating an account I tried to navigate to the Profile page but got the error:

Invalid object name 'dbo.TokenCacheEntries'.
![image](https://cloud.githubusercontent.com/assets/1738479/6049826/e21ef502-ac88-11e4-9b5a-e80625350694.png)

**Comment by dstrockis** [comment]
Ok, I'll have a look. Thanks Cody.  In the meantime to unblock yourself you can probably just delete any remnants of the existing database in App_Data and try running it again.  The App_Data folder should not be included in this repo.

**Comment by dstrockis** [comment]
Also try running code first migrations after deleting the mdf

**Comment by codymullins** [comment]
Yes, I actually did get this to work by deleting the original mdf file. However I was unable to login (MS had issues signing me in) using this project. That is a different issue, though.

**Comment by dstrockis** [comment]
Cool. I'll close this one, feel free to follow up with me if you want to debug the login issue.

**Comment by dstrockis** [comment]
Or open up another issue, that works too :)

**Comment by codymullins** [comment]
@dstrockis I'm actually not sure if this is the correct sample for me to follow. My application is existing and uses forms authentication to authenticate with our on premises active directory. I am trying to pull information from Office 365 & SharePoint.

**Comment by skwan** [comment]
Yes, this is the right sample for that scenario.

**Comment by codymullins** [comment]
Thanks. I'm still having issues with logging in - I'll double check all the configurations I've changed are correct and go through the walk though again before opening another issue.

**Comment by dbarthelemy** [comment]
Unfortunately in my case, when I delete the mdf file or the complete App_Data folder, I got another exception upon Register or Log-in attempts.

System.Data.SqlClient.SqlException: Cannot attach the file 'C:\Users\David\Documents\GitHub\AzureADSamples\WebApp-WebAPI-OAuth2-UserIdentity-DotNet\WebApp\App_Data\aspnet-WebApp-20140401030607.mdf' as database 'aspnet-WebApp-20140401030607'.

This is raised by the line 13.
Ligne 11 :     public class ApplicationDbContext : IdentityDbContext<ApplicationUser>
Ligne 12 :     {
Ligne 13 :         public ApplicationDbContext() : base("DefaultConnection") {}
Ligne 14 : 
Ligne 15 :         public DbSet<TokenCacheEntry> TokenCacheEntries { get; set; }

**Comment by mradamlacey** [comment]
Here's what I did to successfully fix the issue:
1. If you've deleted the MDF, check it out again from the repo:

```
git checkout WebApp/App_Data/*.mdf
git checkout WebApp/App_Data*.ldf
```
1. Open up Tools > NuGet Package Manager > Package Manager Console
2. Enter the following commands:

```
Add-Migration <any name>
Update-Database
```

Explanation - looks like the latest ASP.NET Identity NuGet package needs a db migration to work.  Running Add-Migration will create a migration that will update the local db with the updated model.  `Update-Database` will execute the changes.
